### DISCUSSION TITLE
adding new core generators: sln, global, editorconfig

### DISCUSSION THREAD

**Discussion by peterblazejewicz** [nx-dotnet/nx-dotnet#167] [Ideas]
Hey,

I'd like to somehow help this project (I'm using both nx, dotnet at work and in other places).
May I re-route generator to also create those:
- sln (this would allow to better compose within worskpace for restore, buid, etc)
- global (to fix runtime version)
- editorconfig (to compose with nx workspace configuration, so this one is per-project)
- add to sln (similar to prj to prj)

Editorconfig wuld be nice to have in first place, later sln,

thx!

**Comment by AgentEnder** [comment]
Hey Peter. I've thought about the sln generator a lot in the past, and am still on the fence as to what benefits it provides when you are managing the overall monorepo within Nx. The main benefits I see are just the ability to open all of your projects in an IDE easily. Build composition and such should be orchestrated from Nx's side such that we get the benefits of caching and distributed task execution etc.

The editorconfig would make sense, but generally things like that make more sense at the root level vs the 1/project. The per-project files should just be overrides of the workspace root file. Rather than a new generator, the `init` generator should be modified to create a base file at workspace root, and the apps/libs generators should be modified to generate one in the projects. `.editorconfig` files automatically extend files above them in the file tree, so that should be all thats needed.

`global.json` affects the dotnet cli / SDK version that is used for executing the commands, not the framework build version or anything like that. As such, most people never use it. Since we directly invoke `dotnet`, those files should already be respected. I don't think it would be worth adding that to the generator, since its fairly niche, so long as we maintain compat with it.

I'd definitely be ok with PRs to add editorconfig support 😄
